ssh/tailssh/incubator.go

```diff
@@ -186,6 +195,11 @@ func beIncubator(args []string) error {
 	}
 
 	euid := uint64(os.Geteuid())
+	runningAsRoot := euid == 0
+	if runningAsRoot && ia.isShell() && !ia.isSFTP && ia.loginCmdPath != "" {
+		// If we are trying to launch a login shell, just exec into login instead.
+		return unix.Exec(ia.loginCmdPath, []string{ia.loginCmdPath, "-f", ia.localUser, "-h", ia.remoteIP}, os.Environ())
```

Comment:
Still doesn't preserve the os.Environ() you pass it, `login` nukes the environment by default. If you want to preserve it, need to pass `-p`

Reply:
Speaking of that, where did the env sanitization go? We still want SSH_CONNECTION and SSH_CLIENT envvars to be passed through, but this is just passing through all of os.Environ(), which still contains PATH and stuff, I think?

Reply:
Basically I'm worried about if we're strictly controlling what ends up in this environment. Is it exactly SSH_* (generated by us) plus the allowlisted envs that the client can set before forking the shell?

Reply:
>Speaking of that, where did the env sanitization go? We still want SSH_CONNECTION and SSH_CLIENT envvars to be passed through, but this is just passing through all of os.Environ(), which still contains PATH and stuff, I think?

This happens in `tailscaled`, the env passed down to the `incubator` is always sanitized.

Reply:
Still not passing `-p` to the login process, so it's still going to nuke the entire parent environment.

Reply:
Done